src/pages/workspace/workflows/WorkspaceWorkflowsPage.tsx

```diff
@@ -61,24 +61,33 @@ function WorkspaceWorkflowsPage({policy, betas, route}: WorkspaceWorkflowsPagePr
                           subtitle: translate('workflowsPage.delaySubmissionDescription'),
                           onToggle: (isEnabled: boolean) => {
                               Policy.setWorkspaceAutoReporting(route.params.policyID, isEnabled);
+                              // If the Collect policy has just been created, it will have the frequency set to instant.
+                              // When the user toggles the delayed submission for the first time, we should default to weekly instead.
+                              if (policy?.autoReportingFrequency === CONST.POLICY.AUTO_REPORTING_FREQUENCIES.INSTANT) {
+                                  Policy.setWorkspaceAutoReportingFrequency(route.params.policyID, CONST.POLICY.AUTO_REPORTING_FREQUENCIES.WEEKLY);
```

Comment:
This will make a 2nd API call, won't it? We should combine this logic into `setWorkspaceAutoReporting` if `isEnabled = true` so we only make one API call when enabling the feature.

So basically [here](https://github.com/Expensify/App/blob/a36b0fc53172c9970ec17c3e3f015d7b084eeebf/src/libs/actions/Policy.ts#L416) we should have something like:

```
autoReportingFrequency: enabled ? CONST.POLICY.AUTO_REPORTING_FREQUENCIES.WEEKLY : CONST.POLICY.AUTO_REPORTING_FREQUENCIES.ISNTANT,
```

and then mirror that behavior in the API when calling `setWorkspaceAutoReporting`. We would only ever call `setWorkspaceAutoReportingFrequency` when the feature is already enabled and we just want to change the frequency by itself

Reply:
It will only do this API call the first time and never again after.

If we don't do it, the `Weekly` frequency will never be set as the default value.

Instead the `Instant` frequency will remain?

Reply:
>It will only do this API call the first time and never again after.
Yea, I think we need to combine these calls. I think we can update `setWorkspaceAutoReporting` in App and in the API to also set the frequency if we're turning the feature on AND the current frequency is instant.